src/input/source/led/multicolor_chassis.rs

```diff
@@ -0,0 +1,167 @@
+use crate::{
+    config::LedFixedColor,
+    input::source::{SourceInputDevice, SourceOutputDevice},
+    udev::device::UdevDevice,
+};
+use std::{error::Error, fmt::Debug, path::PathBuf};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum MultiColorChassisError {
+    #[error("Error reading multi_index: {0}")]
+    MultiIndexError(std::io::Error),
+
+    #[error("File multi_intensity not found: {0}")]
+    MultiIntensityNotFound(PathBuf),
+
+    #[error("Error updating multi_intensity: {0}")]
+    MultiIntensityUpdateError(std::io::Error),
+
+    #[error("Unsupported index type: {0}")]
+    UnsupportedIndexType(String),
+}
+
+enum IndexType {
+    RGB,
+}
+
+/// MultiColorChassis source device implementation
+pub struct MultiColorChassis {
+    multi_intensity_path: PathBuf,
+    multi_index_map: Vec<IndexType>,
+    fixed_color: Option<LedFixedColor>,
+}
+
+impl MultiColorChassis {
+    /// Create a new MultiColorChassis source device with the given udev
+    /// device information
+    pub fn new(
+        device_info: UdevDevice,
+        fixed_color: Option<LedFixedColor>,
+    ) -> Result<Self, Box<dyn Error + Send + Sync>> {
+        let multi_intensity_path =
+            PathBuf::from(device_info.syspath().as_str()).join("multi_intensity");
+        if !multi_intensity_path.exists() {
+            return Err(Box::new(MultiColorChassisError::MultiIntensityNotFound(
+                multi_intensity_path,
+            )));
+        }
+
+        let multi_index = PathBuf::from(device_info.syspath().as_str()).join("multi_index");
+        let contents = std::fs::read_to_string(multi_index)
+            .map_err(|err| Box::new(MultiColorChassisError::MultiIndexError(err)))?;
+        let multi_index_strings = contents
+            .split_whitespace()
+            .into_iter()
+            .map(|str| String::from(str))
+            .collect::<Vec<String>>();
```

Comment:
To make clippy happy:

```suggestion
        let multi_index_strings = contents
            .split_whitespace()
            .map(String::from)
            .collect::<Vec<String>>();
```